tracing-subscriber/src/fmt/fmt_subscriber.rs

```diff
@@ -680,12 +680,16 @@ impl<C, N, E, W> Subscriber<C, N, E, W> {
 
 impl<C> Default for Subscriber<C> {
     fn default() -> Self {
+        // only enable ANSI when the feature is enabled, and the NO_COLOR
+        // environment variable is unset or empty.
```

Comment:
it would be nice if the documentation for ANSI color support discussed the behavior with `NO_COLOR`. In particular, it should be explicitly stated somewhere that:
- When the "ansi" feature flag is enabled, ANSI colors are enabled by default if `NO_COLOR` is not set.
- When the `with_ansi` method is called with the value `true`, ANSI colors are _forcibly_ enabled regardless of whether `NO_COLOR` is set.

and, we should document that _any_ non-empty value for `NO_COLOR` will disable ANSI color support.

Reply:
Added a paragraph explaining this behavior to `with_ansi`.